### ISSUE TITLE
tree-sitter 0.20.8 regression build failure against emscripten 3.1.38

### ISSUE THREAD

**Issue by chenrui333** [tree-sitter/tree-sitter#2260]
👋 while trying to upgrade emscripten to 3.1.38, we ran into some regression build failure with 0.20.8 release

The error log is below

```
  error: undefined symbol: $preloadPlugins (referenced by $registerWasmPlugin__deps: ['$preloadPlugins'], referenced by $preloadedWasm__deps: ['$registerWasmPlugin'], referenced by $loadDynamicLibrary__deps: ['$LDSO','$loadWebAssemblyModule','$isInternalSym','$mergeLibSymbols','$newDSO','$asyncLoad','$preloadedWasm'], referenced by $loadDylibs__deps: ['$loadDynamicLibrary','$reportUndefinedSymbols'], referenced by top-level compiled C/C++ code)
  warning: To disable errors for undefined symbols use `-sERROR_ON_UNDEFINED_SYMBOLS=0`
  warning: preloadPlugins may need to be added to EXPORTED_FUNCTIONS if it arrives from a system library
```

Build log, https://github.com/Homebrew/homebrew-core/actions/runs/4953445409/jobs/8877051765?pr=130704
relates to https://github.com/Homebrew/homebrew-core/pull/130704

**Comment by sogaiu** [comment]
IIUC, the currently supported version of emsdk is 3.1.35.

See [here](https://github.com/sogaiu/ts-questions/blob/master/questions/which-version-of-emscripten-should-be-used-for-the-playground/README.md#versions) for some more detail, but [this](https://github.com/tree-sitter/tree-sitter/commits/master/cli/emscripten-version) is where I typically start looking for which version currently works.

**Comment by ahlinc** [comment]
@chenrui333 It's a bug in Emscripten 3.1.38 https://github.com/emscripten-core/emscripten/pull/19361
Use the 3.1.37 version or wait for a newer version with the fixed bug.

**Comment by ahlinc** [comment]
This will be fixed by emscripten-core/emscripten#19362

**Comment by ahlinc** [comment]
This was fixed on Emscripten side.
